components/openzwave-adapter/src/lib.rs

```diff
@@ -204,12 +204,13 @@ fn ensure_directory<T: AsRef<Path> + ?Sized>(directory: &T) -> Result<(), Error>
 }
 
 impl OpenzwaveAdapter {
-    pub fn init<T: AdapterManagerHandle + Send + Sync + 'static> (box_manager: &Arc<T>, user_path: &str, device: Option<String>) -> Result<(), Error> {
+    pub fn init<T: AdapterManagerHandle + Send + Sync + 'static> (box_manager: &Arc<T>, user_path: &str, devices: Option<String>) -> Result<(), Error> {
 
         try!(ensure_directory(user_path));
 
         let options = InitOptions {
-            device: device,
+            // We treat devices as a whitespace delimited list of device names.
+            devices: devices.map(|s| s.split_whitespace().map(|s| s.to_owned()).collect()),
```

Comment:
I couldn't get anything to compile. .cloned isn't implemented for str
split_whitespace essentially returns an Option<&str> which is why I need to use map. .cloned seems to be associated with .iter, so I'm not quite sure how it can be worked into this.